react/frontpage/components/ShowsGraph.jsx

```diff
@@ -0,0 +1,215 @@
+					{Dates.abbreviatedDay(day)}
+				</span> 
+			);
+		});
+
+		var showBlocks = [];
+
+		var fl1, fl2 = 0;
+
+		var start, end; 
+		for(start=1; start < 24; start ++){
+			for(var j = 0; j < this.props.shows.length; j++){
+				if(this.props.shows[j].time == Dates.availableTimes[start]){
+					fl1 = 1;
+					break;
+				}
+			}
+			if(fl1) {break;}
+		}
+
+		for(end = 23; end > 0; end--){
+			for(var j = 0; j < this.props.shows.length; j++){
+				if(this.props.shows[j].time == Dates.availableTimes[end]){
+					fl2 = 1;
+					break;
+				}
+			}
+			if(fl2) {break;}
+		}
+		//this.findStartEndTimes(startTime, endTime);
+
+		for (var hour = start; hour < end+1; hour++) {
+			var hourString = Dates.availableTimes[hour];
+			showBlocks.push( 
+				<div className="hourBlocks">
+					<p className="timeStyle" style={{width: dayWidth}}>{hourString}</p>
+					{ week.map((day) => {
+						var show = this.state.schedule && this.state.schedule[day][hour];
+						return (
+							<ShowBlock isValidShow={(show && show.title)}
+								isCurrentShow={show && show.id === this.props.currentShowID}
+								isActiveShow={show && show.id === this.props.activeShowID}
+								isSpotlightShow={show && show.id === this.props.spotlightShowID}
+								handleClick={()=>{show && this.props.onShowClick(show)}} />
+						);
+					})}
+				</div>
+			);
+		}
+
+		return (
+			<div className="showsGraph">
+				
+				<div className="colorKey">
+					<p className="colorKeyPara">current show</p>
+					<div className="dotCur"></div>
+					<p className="colorKeyPara">spotlight show</p>
+					<div className="dotSpot"></div>
+					
+				</div>
+
+				{dayTitles}
+				{showBlocks}
+			</div>
+		);
+	}
+});
+
+/*
+Individual show block with rollover action
+
+@prop handleClick: callback for mouse click action
+@prop isValidShow: styling bool -> overwrites isCurrentShow and isSpotlightShow if false
+@prop isCurrentShow: styling bool: this show is currently playing
+@prop isActiveShow: styling bool: user clicked on this show
+@prop isSpotlightShow: styling bool: show is a spotlight feature on website
+*/
+var ShowBlock = React.createClass({
+	handleMouseOver: function() {
+		this.setState({active: true});
+	},
+	handleMouseOut: function() {
+		this.setState({active: false});
+	},
+	handleClick: function() {
+		this.props.handleClick()
+	},
+	render: function() {
+
+		if (!this.props.isValidShow) {
+
+			return (
+				<div style={{width: dayWidth}} className="showBlock">
+					<div className="boringBlockStyle" />
+				</div>
+			);
+		} else {
+			var blockColor = (this.props.isActiveShow && 'red')
+				|| (this.props.isSpotlightShow && 'purple')
+				|| (this.props.isCurrentShow && '#3c84cc')
+				|| 'black';
+
+			return (
+				<div className="showBlock" style={{width: dayWidth}}>
+					<div className="blockStyle" 
+						style={{backgroundColor: blockColor}}
+						onMouseOver={this.handleMouseOver}
+						onMouseOut={this.handleMouseOut}
```

Comment:
are we still using the `onMouseOut` events? I think a roll-over color change is cool, but it doesn't look like that's implemented and this isn't necessary for that